pkg/kubelet/kubelet.go

```diff
@@ -2867,6 +2867,7 @@ func (kl *Kubelet) syncNetworkStatus() {
 
 // Set addresses for the node.
 func (kl *Kubelet) setNodeAddress(node *api.Node) error {
+	hostnameAddress := api.NodeAddress{Type: api.NodeHostName, Address: kl.GetHostname()}
```

Comment:
No, not necessarily: `// GetHostname Returns the hostname as the kubelet sees it.` Also, this can be overriden arbitrarily using `--hostname-override` on the kubelet. Maybe there are cases where this behaviour is undesired, e.g.: cloud-provider sets a resolvable Nodename, but the Node's hostname is actually not resolvable by the master, however it is preferred b/c the Addresses array is populated now. contrived?